### DISCUSSION TITLE
help/contributing/

### DISCUSSION THREAD

**Discussion by glenn-jocher** [ultralytics/ultralytics#8011] [Docs]
# help/contributing/

A bit of text here.

https://docs.ultralytics.com/help/contributing/

**Comment by glenn-jocher** [comment]
Hey there! 🚀 

Excited to explore the world of YOLO models with Ultralytics? You've come to the right place! This Help page is your gateway to resources, guides, and documentation designed to boost your journey. Whether you have questions, want to contribute, or are looking to resolve an issue, we're here to support you. Dive into our FAQs, contribution guidelines, and more to enhance your experience. 

Got something to share or a question on your mind? We'd love to hear from you. Let's collaborate and grow together! Happy coding!

**Comment by esssyjr** [comment]
Good evening all.  I am trying to make a contribution for the dataset. I have forked and created a new branch but I could not see the folder specified for dataset. Should I just submit the .zip file to the root together with the read me, and send a pull request? Also, in my Github desktop I am unable to see Jupyter notebook as a code editor, but I manage to open the ultralytics cloned file in jupyter notebook, and uploaded the dataset. is that a good practice? Sorry for silly questions, I am new to open source contibution.

**Reply by glenn-jocher** [reply]
@esssyjr thank you for your interest in contributing! For dataset contributions, please follow the [Ultralytics Contribution Guide](https://docs.ultralytics.com/help/contributing/) and submit your dataset in a directory with a clear structure, including `train/` and `val/` folders as applicable. Place the README in the same directory. Regarding Jupyter Notebook, it's fine to use it for preparing your dataset, but ensure your final submission is in the proper format per the guidelines. If you have more questions, feel free to ask on GitHub.

**Reply by esssyjr** [reply]
Thanks for the reply, I have successfully uploaded the zip and readme file in the a folder in root of the Ultralytics repository. I requested a pull request in which it pass all the test but merge have been blocked, waiting for a approve review. Thank YOU

**Reply by glenn-jocher** [reply]
Thank you for your contribution! A team member will review your pull request shortly. Please ensure all guidelines are followed for a smooth review process.

**Comment by Mehrab-Shahbazi** [comment]
Hi
I will be happy if yolo add a feature that use slding window object detection. i want to add it and pull request it. is it usefull feature? will you accebt it?(if the tests passes).

**Reply by Mehrab-Shahbazi** [reply]
by sliding window, i mean crop big images ti smaller tiles and at the end merge all results as one output.( it is usefull for big images as you know)

**Reply by glenn-jocher** [reply]
@Mehrab-Shahbazi thank you for your interest in contributing! Sliding window object detection for large images can indeed be useful. If implemented efficiently and the tests pass, we welcome your pull request for review. Please ensure the feature aligns with the repository's design principles. For guidance, refer to [our contributing guide](https://docs.ultralytics.com/help/contributing/).

**Reply by Mehrab-Shahbazi** [reply]
thank for your reply

**Reply by glenn-jocher** [reply]
@Mehrab-Shahbazi you're welcome! If you have any further questions or need assistance with your contribution, feel free to ask.
